airflow/www/api/experimental/endpoints.py

```diff
@@ -99,6 +99,30 @@ def trigger_dag(dag_id):
     return response
 
 
+@api_experimental.route('/dags/<string:dag_id>/dag_runs', methods=['GET'])
```

Comment:
looks like this endpoint already exists below as a way to get all dag runs so we should make this an endpoint specifically for getting a dagrun with the given run ID